cheroot/test/test_ssl.py

```diff
@@ -528,6 +567,15 @@ def test_ssl_env(
             )
         pytest.fail(msg)
 
+    # to perform the ssl handshake over that loopback socket,
+    # the builtin ssl environment generation uses a thread
+    if thread_exceptions:
+        for _, _, trace in thread_exceptions:
+            print(trace, file=sys.stderr)
+        pytest.fail(
+            thread_exceptions[0][0].__name__ + ': ' + thread_exceptions[0][1],
+        )
```

Comment:
I think the pytest-native `assert` idiom would be better suitable here...
```suggestion
    for _, _, trace in thread_exceptions:
        print(trace, file=sys.stderr)
    assert not thread_exceptions, ': '.join((
        thread_exceptions[0][0].__name__,
        thread_exceptions[0][1],
    ))
```